tests/api/common/experimental/connection_tests.py

```diff
@@ -0,0 +1,427 @@
+            self.assertEqual(result, ('new3', 'postgres', 'host', 5432,
+                                      extra['new3']))
+
+    def test_delete_connection(self):
+        session = settings.Session()
+        # add some conns to delete
+        uri = 'postgres://airflow:airflow@host:5432/airflow'
+
+        self.assertEqual(connections.add_connection(
+            conn_id="new1",
+            conn_uri=uri
+        ), Connection(conn_id="new1", uri=uri))
+
+        self.assertEqual(connections.add_connection(
+            conn_id="new2",
+            conn_uri=uri
+        ), Connection(conn_id="new2", uri=uri))
+
+        self.assertEqual(connections.add_connection(
+            conn_id="new2",
+            conn_uri=uri
+        ), Connection(conn_id="new2", uri=uri))
+
+        # delete one
+        self.assertEqual(connections.delete_connection(conn_id='new1'),
+                         "Successfully deleted `conn_id`=new1")
+
+        # try to delete multiple with delete_all=False
+        self.assertEqual(connections.delete_connection(conn_id='new2'),
+                         'Found 2 connection with ' +
+                         '`conn_id`=new2. Specify `delete_all=True` to remove all')
+
+        # make sure none were deleted
+        self.assertEqual(2, len((session.query(models.Connection)
+                                 .filter(models.Connection.conn_id == 'new2').all())))
+
+        self.assertEqual(connections.delete_connection(conn_id='new2', delete_all=True),
+                         "Successfully deleted 2 connections with `conn_id`=new2")
+
+        # make sure none were deleted
+        self.assertEqual(0, len((session.query(models.Connection).all())))
+
+        # Attempt to delete a non-existing connnection
+        self.assertEqual(connections.delete_connection(conn_id='non_existent'),
+                         "Did not find a connection with `conn_id`=non_existent")
+
+        # Attempt to delete a non-existing connnection
+        self.assertRaises(MissingArgument, connections.delete_connection, None)
+
+        session.close()
+
+    def test_list_connections(self):
+
+        uri = 'postgres://airflow:airflow@host:5432/airflow'
+
+        # add a connection using conn_uri
+        self.assertEqual(connections.add_connection(
+            conn_id="new1",
+            conn_uri=uri
+        ), Connection(conn_id="new1", uri=uri))
+
+        # add a connection with all params defined
+        self.assertEqual(connections.add_connection(
+            conn_id="new2",
+            conn_type="postgres",
+            conn_host="host",
+            conn_login="airflow",
+            conn_password="airflow",
+            conn_schema="airflow",
+            conn_port="5432"
+        ), Connection(
+            conn_id="new2",
+            conn_type="postgres",
+            host="host",
+            login="airflow",
+            password="airflow",
+            schema="airflow",
+            port="5432"
+        ))
```

Comment:
From a quick look it appears that all the test cases use this connection - could we create it once in setUp() to reduce the amount of code in each test function?